Dockerfile

```diff
@@ -0,0 +1,166 @@
+ARG unitycatalog_etc="etc"
+ARG unitycatalog_bin="bin"
+ARG unitycatalog_user_name="unitycatalog"
+ARG unitycatalog_user_home="home"
+ARG unitycatalog_user_basedir="${unitycatalog_home}/${unitycatalog_user_home}"
+# Specify any custom parameters necessary to generate
+# the Uber-Jar by SBT. 
+# Note: The default allocated heam memory size is too small
+# and will cause the process to fail when attempting to compile
+# and generate the Uber-Jar. Therefore it is important to choose
+# a size large enough for the compiler to run. 
+ARG sbt_args="-J-Xmx5G"
+
+
+
+# ###### STAGE 1 ###### #
+# Building the Uber-Jar #
+#########################
+
+FROM eclipse-temurin:22-jdk-alpine AS build_uc
+
+ARG unitycatalog_repo
+ARG sbt_args
+
+# Install any required packages
+RUN <<EOF
+    set -ex;
+    apk update;
+    apk upgrade; 
+    apk add bash git;
+    rm -R /var/cache/apk/*;
+EOF
+
+# Create temporary directories for initialization files
+RUN mkdir -p "${unitycatalog_repo}";
+
+WORKDIR "${unitycatalog_repo}"
+
+COPY . .
+
+RUN build/sbt ${sbt_args} server/assembly
+
+
+# ###### STAGE 2 ###### #
+# Running the UC server #
+#########################
+
+FROM eclipse-temurin:22-jre-alpine AS run_uc
+
+ARG unitycatalog_uid
+ARG unitycatalog_home
+ARG unitycatalog_repo
+ARG unitycatalog_jar
+ARG unitycatalog_etc
+ARG unitycatalog_bin
+ARG unitycatalog_user_name
+ARG unitycatalog_user_home
+ARG unitycatalog_user_basedir
+ARG sbt_args
+
+EXPOSE 8081
+
+RUN <<EOF
+    set -ex;
+    apk update;
+    apk upgrade; 
+    apk add bash;
+    rm -R /var/cache/apk/*;
+EOF
+
+# Define the shell used within the container
+SHELL ["/bin/bash", "-i", "-c", "-o", "pipefail"] 
+
+ENV SERVER_TARGET_DIR="${unitycatalog_home}/${unitycatalog_jar}"
+ENV UC_SERVER_BIN="${unitycatalog_home}/${unitycatalog_bin}/start-uc-server"
+
+# Create the UC directories
+RUN <<-EOF 
+    set -ex;
+    mkdir -p "${unitycatalog_home}";
+    mkdir -p "${unitycatalog_home}/${unitycatalog_jar}";
+    mkdir -p "${unitycatalog_home}/${unitycatalog_etc}";
+    mkdir -p "${unitycatalog_home}/${unitycatalog_bin}";
+    mkdir -p "${unitycatalog_home}/${unitycatalog_user_home}";
+EOF
+
+
+# Create system group and user for Unity Catalog
+# ENsure the user created has their HOME pointing to the volume
+# created to persist user data and the sbt cached files that 
+# are created as a result of compiling the unity catalog.
+# This also ensures that the container can run independently from
+# the storage, so we can have ephemeral docker instances with --rm
+# and still be able to run the unity catalog each time without problems.
+RUN <<-EOF
+    #!/usr/bin/env bash
+    set -ex;
+    addgroup --system --gid "${unitycatalog_uid}" "${unitycatalog_user_name}";
+    adduser --system --uid "${unitycatalog_uid}" \
+            --ingroup "${unitycatalog_user_name}" \
+            --home "${unitycatalog_user_basedir}" \
+            --shell "$(/usr/bin/env bash)" \
+            "${unitycatalog_user_name}";
+EOF
```

Comment:
what is this needed for? I would like the initial setup to be as minimal as possible without losing features.

Reply:
It's important to run the catalogue as a less privileged user rather than using root.
This line creates a user that owns the catalog and changes the default home directory of the user to an externally mounted volume so that data is persisted when the docker container restarts.